evm_loader/program/src/account_storage.rs

```diff
@@ -243,8 +242,22 @@ impl<'a> ProgramAccountStorage<'a> {
                         debug_print!("Apply can't be done. Not found account for address = {:?}.", address);
                         return Err(ProgramError::NotEnoughAccountKeys);
                     }
+                },
+                Apply::Delete { address } => {
+                    debug_print!("Going to delete address = {:?}.", address);
+
+                    if let Some(pos) = self.find_account(&address) {
+                        let account_info = &self.account_metas[pos];
+                        debug_print!("Move funds from account");
+                        **account_info.lamports.borrow_mut() = 0;
```

Comment:
You should transfer these funds to some account. In another case post transaction check fails.